packages/account/src/account.ts

```diff
@@ -569,6 +572,90 @@ export class Account extends AbstractAccount {
     return this.sendTransaction(request);
   }
 
+  async getTransactionCost(
+    transactionRequestLike: TransactionRequestLike,
+    { signatureCallback, quantitiesToContract = [] }: TransactionCostParams = {}
+  ): Promise<TransactionCost> {
+    const txRequestClone = clone(transactionRequestify(transactionRequestLike));
+    const isScriptTransaction = txRequestClone.type === TransactionType.Script;
+    const baseAssetId = this.provider.getBaseAssetId();
+
+    // Fund with fake UTXOs to avoid not enough funds error
+    // Getting coin quantities from amounts being transferred
+    const coinOutputsQuantities = txRequestClone.getCoinOutputsQuantities();
+    // Combining coin quantities from amounts being transferred and forwarding to contracts
+    const allQuantities = mergeQuantities(coinOutputsQuantities, quantitiesToContract);
+    // Funding transaction with fake utxos
+    txRequestClone.fundWithFakeUtxos(allQuantities, baseAssetId, this.address);
+
+    /**
+     * Estimate predicates gasUsed
+     */
+    // Remove gasLimit to avoid gasLimit when estimating predicates
+    if (isScriptTransaction) {
+      txRequestClone.gasLimit = bn(0);
+    }
+
+    /**
+     * The fake utxos added above can be from a predicate
+     * If the resources owner is a predicate,
+     * we need to populate the resources with the predicate's data
+     * so that predicate estimation can happen.
+     */
+    if (this && 'populateTransactionPredicateData' in this) {
+      (this as unknown as Predicate<[]>).populateTransactionPredicateData(txRequestClone);
+    }
+
+    const txCost = await this.provider.getTransactionCost(txRequestClone, {
+      signatureCallback,
+      funded: true,
+    });
+
+    return {
+      ...txCost,
+      requiredQuantities: allQuantities,
+    };
+  }
+
+  /**
+   * Get the required quantities and associated resources for a transaction.
+   *
+   * @param transactionRequestLike - transaction request to populate resources for.
+   * @param quantitiesToContract - quantities for the contract (optional).
+   *
+   * @returns a promise resolving to the required quantities for the transaction.
+   */
+  async getResourcesForTransaction(
+    transactionRequestLike: TransactionRequestLike,
+    quantitiesToContract: CoinQuantity[] = []
+  ) {
```

Comment:
What is the motivation for moving this method to the Account class?

Reply:
@petertonysmith94 I was just curious about this. I am under the impression that we can remove `getResourcesForTransaction` entirely.